java/arcs/core/data/Plan.kt

```diff
@@ -21,8 +21,33 @@ import arcs.core.util.lens
 open class Plan(
     // TODO(cromwellian): add more fields as needed (e.g. RecipeName, etc for debugging)
     val particles: List<Particle>,
-    val arcId: String? = null
+    val annotations: List<Annotation>
```

Comment:
Note that when you add fields to Plan, if you ever expect to use them outside of Allocator (e.g. in ArcHosts), you need to update ParcelablePlan/PlanPartition.

Reply:
thanks, that's good to know. 
this is just a first of prs with regards to annotations support, so i'll make this in a follow up, if that's ok?